storage/remote/write_handler.go

```diff
@@ -250,7 +250,7 @@ func (h *writeHandler) write(ctx context.Context, req *prompb.WriteRequest) (err
 
 		// TODO(bwplotka): Even as per 1.0 spec, this should be a 400 error, while other samples are
 		// potentially written. Perhaps unify with fixed writeV2 implementation a bit.
-		if !ls.Has(labels.MetricName) || !ls.IsValid(model.NameValidationScheme) {
+		if !ls.Has(labels.MetricName) || !ls.IsValid(model.UTF8Validation) {
```

Comment:
this is the only bit I'm worried about -- I think this just means that Prometheus will not check metrics before trying to remote write them, and if the other end does not support UTF-8 it will return an error. Would appreciate a second look here. (and below)

Reply:
Not sure if I understand the worry. This changes nothing for Prometheus (relaxed validation was always there due to global value). Plus this is the receiving/ingest code, do you mean client side? I'm also pretty confident this is rarely imported code downstream -- thus we should be safe.

One could vote for some explicit validation option, like on scrape, but I would wait for someone to report the need for this.

Thanks for double checking!